src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -44,9 +44,17 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
                 //if (featureFlag.Conditions?.ClientFilters == null)
                 if (featureFlag.Conditions?.ClientFilters == null || !featureFlag.Conditions.ClientFilters.Any()) // workaround since we are not yet setting client filters to null
                 {
-                    //
-                    // Always on
-                    keyValues.Add(new KeyValuePair<string, string>(featureFlagPath, true.ToString()));
+                    // Add the AlwaysOn filter instead of setting the flag to "true" so feature management doesn't skip evaluating variants and telemetry when present
+                    if (featureFlag.Variants != null || featureFlag.Telemetry != null)
```

Comment:
Just want to mention that this condition could be removed so that we just always send `AlwaysOn` instead of setting the flag to `true`. It simplifies the logic and would better handle any future additions to the feature definition, but it would basically remove [the optimization](https://github.com/microsoft/FeatureManagement-Dotnet/blob/9a7f2ffcf6648426a66f3983ca15b5948ada154b/src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs#L216) in the feature management library to skip evaluating the rest of the flag if it's set to true.

Reply:
I prefer removing the optimization for simplicity / if we add more fields in the future, we don't have to update this.

Reply:
I agree on simplicity.